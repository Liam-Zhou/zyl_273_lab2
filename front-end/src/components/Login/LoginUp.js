import React, {Component} from 'react';
import '../../App.css';
import { connect } from "react-redux";
import axios from 'axios';
import cookie from 'react-cookies';
import {Redirect} from 'react-router';
import config from '../../config/basicConfig'

//Define a Login Component
class LoginUp extends Component{
    //call the constructor method
    constructor(props){
        //Call the constrictor of Super class i.e The Component
        super(props);
        //const mes = props.location.state
        this.state = {
            email : "",
            password : "",
            role : "",
            message:''
        }
        //Bind the handlers to this class
        this.emailChangeHandler = this.emailChangeHandler.bind(this);
        this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
        this.submitLogin = this.submitLogin.bind(this);
        this.roleChangeHandler = this.roleChangeHandler.bind(this);
    }
    componentWillMount(){
    }

    emailChangeHandler = (e) => {
        this.setState({
            email : e.target.value
        })
    }
    //password change handler to update state variable with the text entered by the user
    passwordChangeHandler = (e) => {
        this.setState({
            password : e.target.value
        })
    }

    roleChangeHandler = (e) => {
        this.setState({
            role : e.target.value
        })
    }
    //submit Login handler to send a request to the node backend
    submitLogin = (e) => {
        //prevent page from refresh
        e.preventDefault();
        
        if(this.state.email && this.state.password && this.state.role){
            
            const data = {
                email : this.state.email,
                password : this.state.password,
                role : this.state.role
            }
            let host = config.host;
            let port = config.back_end_port;
            let url = host + ':' + port;

            axios.defaults.withCredentials = true;
            //make a post request with the user data
            axios.post(url + '/login/',data)
                .then(response => {
                    console.log("Status Code : ",response.status);
                    if(response.status === 200){
                        console.log('response.data',response.data)
                        if(response.data == 'noinfo'){
                                this.setState({
                                    message : "no such account!"
                        })
                    }
                        if(response.data == 'mysql error'){
                            this.setState({
                                message : "sorry database error!"
                        })
                    }
                        if(response.data.id){
                            let data = JSON.stringify(response.data)
                            sessionStorage.setItem('user',data)
                            this.setState({
                                email : "",
                                password : "",
                                role : '',
                                message : 'success'
                            })

                        }
                    }else{
                        
                    }
                });
            }else{
                this.setState({
                    message : "no empty please!"
                })
            }
    }

    render(){
        let redirectVar = null;
        if(cookie.load('cookie')||sessionStorage.getItem('user')){
            let role = JSON.parse(sessionStorage.getItem('user')).role;
            if(role == 'student'){
                redirectVar = <Redirect to= "/job/jobSearch"/>
            }
            if(role == 'company'){
                redirectVar = <Redirect to= "/job/viewJob"/>
            }
            
        }
        // if(sessionStorage.getItem('user')){
        //     redirectVar = <Redirect to= "/"/>
        // }
        return(
            <div class="loginform">
                        <div>
                            {redirectVar}
                        <div align ="center">
                            <p>Please enter your information</p>
                        </div>
                            <div class="form-group">
                                <input onChange = {this.emailChangeHandler}  type="text" class="form-control" name="email" placeholder="Email"/>
                            </div>
                            <div class="form-group">
                                <input onChange = {this.passwordChangeHandler}  type="password" class="form-control" name="password" placeholder="Password"/>
                            </div>
                            <label class="radio-inline">
                                <input type="radio" name="role"  onChange={this.roleChangeHandler} value="student"/> student
                            </label>
                            <label class="radio-inline">
                                <input type="radio" name="role"  onChange={this.roleChangeHandler} value="company"/> company
                            </label>
                            <button onClick = {this.submitLogin} class="button">Login</button> 
                            </div>  
                            <div><h4>{this.state.message}</h4></div>  
            </div>            
                            
        )
    }
}
export default LoginUp;