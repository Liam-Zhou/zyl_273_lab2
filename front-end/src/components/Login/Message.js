import React from "react";
import { connect } from "react-redux";
const messageShow = ({ message }) => (
    <div><h4>{message}</h4></div>
);
const mapStateToProps = state => {
  console.log('state in message',state)
  return { message: state.message };
};
const Message = connect(mapStateToProps)(messageShow);
export default Message;