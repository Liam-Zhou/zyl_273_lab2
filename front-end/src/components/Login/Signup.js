import React, {Component} from 'react';
import '../../App.css';
import axios from 'axios';
import cookie from 'react-cookies';
import {Redirect} from 'react-router';
import config from '../../config/basicConfig'

//Define a Signup Component
class Signup extends Component{
    //call the constructor method
    constructor(props){
        //Call the constrictor of Super class i.e The Component
        super(props);
        //maintain the state required for this component
        this.state = {
            emailId : "",
            password : "",
            name : "",
            location : "",
            collegeName : "",
            role : "",
            hidden1 : "hidden",
            hidden2 : "hidden",
            message : null,
            lead : ''
        }
        //Bind the handlers to this class
        this.nameChangeHandler = this.nameChangeHandler.bind(this);
        this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
        this.emailIdChangeHandler = this.emailIdChangeHandler.bind(this);
        this.locationChangeHandler = this.locationChangeHandler.bind(this);
        this.collegeNameChangeHandler = this.collegeNameChangeHandler.bind(this);
        this.roleChangeHandler = this.roleChangeHandler.bind(this);

        this.submitInfo = this.submitInfo.bind(this);
    }
    //Call the Will Mount to set the auth Flag to false
    componentWillMount(){
    }

    //username change handler to update state variable with the text entered by the user
    nameChangeHandler = (e) => {
        this.setState({
            name : e.target.value
        })
    }
    //password change handler to update state variable with the text entered by the user
    passwordChangeHandler = (e) => {
        this.setState({
            password : e.target.value
        })
    }
    emailIdChangeHandler = (e) => {
        this.setState({
            emailId : e.target.value
        })
    }
    locationChangeHandler = (e) => {
        this.setState({
            location : e.target.value
        })
    }
    collegeNameChangeHandler = (e) => {
        this.setState({
            collegeName : e.target.value
        })
    }


    roleChangeHandler = (e) => {
        let role = e.target.value
        this.setState({
            role : role
        })
        
        if(role == 'student'){
            this.setState({
                hidden1 : "text",
                hidden2 : "hidden",
                location : ''
            })
        }else{
            this.setState({
                hidden1 : "hidden",
                hidden2 : "text",
                collegeName : ''
            })
        }
    }
    //submit Login handler to send a request to the node backend
    submitInfo = (e) => {
        //prevent page from refresh
        e.preventDefault();
        if(this.state.name && this.state.password && this.state.role && this.state.emailId){
            
            const data = {
                name : this.state.name,
                password : this.state.password,
                emailId : this.state.emailId,
                location : this.state.location,
                collegeName : this.state.collegeName,
                role : this.state.role
            }
            let host = config.host;
            let port = config.back_end_port;
            let url = host + ':' + port;
            //make a post request with the user data
            axios.post(url + '/signup/',data)
                .then(response => {
                    console.log("Status Code : ",response.status);
                    if(response.status === 200){
                        console.log('response.data',response.data)
                        if(response.data != 'success'){
                            this.setState({
                                message : response.data
                            })
                        }else{
                            //this.props.history.push('/login','sign up success!!')
                            alert('sign up success!!')
                            this.setState({
                                lead : <Redirect to= "/login"/>
                            })
                            
                        }
                    }else{
                        
                    }
                });
            }else{
                this.setState({
                    message : "no empty please!"
                })
            }
    }

    render(){
        //redirect based on successful login
        let host = config.host;
        let port = config.front_end_port;
        let url = host + ':' + port;
        return(
            <div>
                {this.state.lead}
            <div class="container">
                
                <div class="maincenter">
                    <div class="loginform">
                        <div align ="center">
                            <p>Please register your information</p>
                        </div>
                            <div class="form-group">
                               <input onChange = {this.nameChangeHandler}  type="text" class="form-control" name="name" placeholder="name"/>
                            </div>
                            <div class="form-group">
                                <input onChange = {this.emailIdChangeHandler}  type="text" class="form-control" name="email" placeholder="EmailId"/>
                            </div>
                            <div class="form-group">
                                <input onChange = {this.passwordChangeHandler}  type="password" class="form-control" name="password" placeholder="Password"/>
                            </div>
                            <div class="form-group" >
                                <input onChange = {this.locationChangeHandler} type = {this.state.hidden2}  class="form-control" name="location" placeholder="Location"/>
                            </div>
                            <div class="form-group" >
                                <input onChange = {this.collegeNameChangeHandler} type = {this.state.hidden1} class="form-control" name="collegename" placeholder="College name"/>
                            </div>


                            <label class="radio-inline">
                                <input type="radio" name="role"  onChange={this.roleChangeHandler} value="student"/> student
                            </label>
                            <label class="radio-inline">
                                <input type="radio" name="role"  onChange={this.roleChangeHandler} value="company"/> company
                            </label>
                            <button onClick = {this.submitInfo} class="button">Signup</button>                 
                            <div><h4>{this.state.message}</h4></div>
                        </div>
                        <div class='signin'>
                        <p >already have an account? <a href={url + '/login'} class="navbar-link">signin</a></p>
                    </div>
                </div>
            </div>
            </div>
        )
    }
}

export default Signup;