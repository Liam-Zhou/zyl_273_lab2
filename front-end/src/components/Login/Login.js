import React, {Component} from 'react';
import '../../App.css';
import axios from 'axios';
import cookie from 'react-cookies';
import {Redirect} from 'react-router';
import config from '../../config/basicConfig'
import LoginUp from'./LoginUp'

//Define a Login Component
class Login extends Component {
    render() {
        let host = config.host;
        let port = config.front_end_port;
        let url = host + ':' + port;
        return(
            <div>
            <div class="container">
                
                <div class="maincenter">
                    <div align ="center">
                        <img class='img' src={require('../../img/handshake.png')} alt="." />
                    </div>
                    
                        <LoginUp/>
                        
                    <div class='signup'>
                        <p class="signup">Don't have an account? <a href={url + '/signup'} class="navbar-link">signup</a></p>
                    </div>
                </div>
            </div>
            </div>
        )
    }
}
//export Login Component
export default Login;