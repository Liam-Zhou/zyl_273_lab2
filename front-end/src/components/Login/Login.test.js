
import React from 'react';
import Login from './Login';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('login page correctly', () => {
  const component = mount( <Login ></Login>);
  expect(component).toMatchSnapshot();
});
