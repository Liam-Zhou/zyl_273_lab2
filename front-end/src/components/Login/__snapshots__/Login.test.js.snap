// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page correctly 1`] = `ReactWrapper {}`;
